dm/worker/subtask_holder.go

```diff
@@ -46,6 +46,18 @@ func (h *subTaskHolder) removeSubTask(name string) {
 	delete(h.subTasks, name)
 }
 
+// resetAllSubTasks does Close, change cfg.UseRelay then Init the subtasks
+func (h *subTaskHolder) resetAllSubTasks(useRelay bool) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	for _, st := range h.subTasks {
+		stage := st.Stage()
+		st.Close()
+		st.cfg.UseRelay = useRelay
```

Comment:
Should we compare `st.cfg.UseRelay` with `useRelay`?

Reply:
I want to switch syncer's binlog type 🤔 why compare it

Reply:
I mean if `st.cfg.UseRelay` has already been equal to `useRelay`, do we still need to restart this subtask?